### PR TITLE
Feature/ab#898 preselect classification

### PR DESCRIPTION
**Pull request by flostyl3**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by flonix8** [comment]
It might be a good idea, to change the DemoSite such that one of the demo classes is automatically selected.
(In case we're still maintaining that demo. Otherwise nevermind)

File: src/ClassSelectionMenu/index.js [hunk 1]
```diff
@@ -70,8 +71,11 @@ export const ClassSelectionMenu = ({
       }
     }
     window.addEventListener("keydown", onKeyDown)
+    if(preselectCls!=null && selectedCls==null){
```
Comment (flonix8) [inline src/ClassSelectionMenu/index.js]: Some more whitespace would be nice :)

File: src/MainLayout/index.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ export const MainLayout = ({
           ({
             type,
             ...params.reduce(
-              (acc, p, i) => ((acc[p] = args[i]), acc), {})
+              (acc, p, i) => (((acc[p] = args[i]), acc)), {})
```
Comment (flonix8) [inline src/MainLayout/index.js]: I have absolutely no idea what this does :sweat_smile: 
Are you sure, the parenthesis change anything?

**Comment by flonix8** [comment]
Also, we might want to add the new property to the list of available props in `README.md`

**Review by flonix8 — CHANGES_REQUESTED** [review]

**Comment by flonix8** [comment]
I had some more things that I wanted you to check (not sure if they actually have to be changed)

> It might be a good idea, to change the DemoSite such that one of the demo classes is automatically selected. (In case we're still maintaining that demo. Otherwise nevermind)

Please also have a look at my previous comment.

**Comment by flostyl3** [comment]
> It might be a good idea, to change the DemoSite such that one of the demo classes is automatically selected. (In case we're still maintaining that demo. Otherwise nevermind)

Good suggest. i added this

**Comment by flostyl3** [comment]
> Also, we might want to add the new property to the list of available props in `README.md`

fixed it

File: src/MainLayout/index.js (hunk 2 above)
Comment (flostyl3) [reply src/MainLayout/index.js]: there where compiler Warnings about this so I fixed it. I don't know what the difference is ether

File: src/ClassSelectionMenu/index.jsx [hunk 3]
```diff
@@ -54,9 +54,33 @@ const Number = styled("div")(({theme}) => ({
 
 export const ClassSelectionMenu = ({
   selectedCls,
+  preselectCls,
   regionClsList,
   onSelectCls,
 }) => {
+  useEffect(() => {
```
Comment (flonix8) [inline src/ClassSelectionMenu/index.jsx]: I have now found out why this useEffect() call had been removed in the meantime (and we readded it earlier during the merge).
This is part of the problematic keyboard shortcuts, that Anett removed. So you have to remove all key related parts and check if the implementation still works.

**Review by flonix8 — CHANGES_REQUESTED** [review]

**Pull request merged**